pkg/runtime/cache/cache.go

```diff
@@ -14,33 +14,48 @@
 package cache
 
 import (
-	"os"
 	"time"
 
 	"github.com/go-logr/logr"
+	"github.com/jaypipes/envutil"
 	kubernetes "k8s.io/client-go/kubernetes"
 )
 
 const (
-	// defaultNamespace is the default namespace to use if the environment
-	// variable NAMESPACE is not found. The NAMESPACE variable is injected
-	// using the kubernetes downward api.
-	defaultNamespace = "ack-system"
+	// envVarACKSystemNamespace is the string key for the environment variable
+	// storing the Kubernetes Namespace we use for ConfigMaps and other ACK
+	// system configuration needs.
+	envVarACKSystemNamespace = "ACK_SYSTEM_NAMESPACE"
+
+	// envVarDeprecatedK8sNamespace is the string key for the old, deprecated
+	// environment variable storing the Kubernetes Namespace we use for
+	// ConfigMaps and other ACK system configuration needs.
+	envVarDeprecatedK8sNamespace = "K8S_NAMESPACE"
```

Comment:
We should also change `K8S_NAMESPACE` in https://github.com/aws-controllers-k8s/code-generator/blob/00575997b75078c641290c7d0ec2a9fedfbdabed/templates/helm/templates/deployment.yaml#L59-L63 and https://github.com/aws-controllers-k8s/code-generator/blob/6da05cd784c388f3b113778d034960398c2dd56e/templates/config/controller/deployment.yaml.tpl